xmodule/capa_module.py

```diff
@@ -685,7 +685,9 @@ def generate_report_data(self, user_state_iterator, limit_responses=None):
             anonymous_student_id=None,
             cache=None,
             can_execute_unsafe_code=lambda: None,
-            get_python_lib_zip=(lambda: get_python_lib_zip(contentstore, self.runtime.course_id)),
+            get_python_lib_zip=(
+                lambda: SandboxService(contentstore, self.scope_ids.usage_id.context_key).get_python_lib_zip()
```

Comment:
Was this tested manually? This looks perfectly reasonable to me, but sandboxed code execution edge cases have bitten us a lot over the years.

Reply:
Old Mongo courseware is no longer accessible either by LMS or Studio, so let's not worry about its behavior w.r.t. this.